client/src/quit/components/QuitButton.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import {  Button, Header, Icon, Modal  } from "semantic-ui-react";
+import { handleSubmit } from "../api"
+
+
+class QuitButton extends React.Component{
+    public state = { modelOpen: false }
+
+    public handleOpen = () => {
+        this.setState({ modelOpen: true })
+    }
+
+    public handleClose = () => {
+        this.setState({ modelOpen: false })
+    }
+
+    public handleQuit = () => {
+        this.handleClose()
+        handleSubmit()
+    }
+
+    public render(){
+    return(
+        <Modal
+            trigger={<Button content="Quit" icon="shutdown" onClick={this.handleOpen} labelPosition="left" floated="right" />} 
+            open={this.state.modelOpen}
+            onClose={this.handleClose}
+            size='mini'
+        >
+            <Header icon='shutdown' content='Confirm Exit' />
+            <Modal.Content>
+                <p>
+                    Do you want to exit ?
+                </p>
+            </Modal.Content>
+            <Modal.Actions>
+                <Button color='red' onClick={this.handleClose}>
+                    <Icon name='remove'/> No
+                </Button>
+                <Button color='green' onClick={this.handleQuit} >
```

Comment:
About button colors and labels: instead of red/green, the convention of the rest of the UI should be followed: the "yes" button should be marked as "primary" (then it will automatically become blue), the secondary action should stay default (grey). That way the buttons all behave consistently. Instead of putting "yes" and "no" as labels of the button, the labels could be "Shutdown" (or "Exit") and "Cancel" - because user's don't like to read, they can just click on the action they want to execute, instead having to read and understand the text above.

I know this may seem very minor, but things like this add up to a more fluid user experience.